src/mem/remoteallocator.h

```diff
@@ -106,9 +106,12 @@ namespace snmalloc
       invariant();
       freelist::Object::atomic_store_null(last, key);
 
-      // exchange needs to be a release, so nullptr in next is visible.
+      // Exchange needs to be acq_rel.
+      // *  It needs to be a release, so nullptr in next is visible.
+      // *  Needs to be acquire, so linking into the list does not race with
+      //    the other threads nullptr init of the next field.
       freelist::QueuePtr prev =
-        back.exchange(capptr_rewild(last), std::memory_order_release);
+        back.exchange(capptr_rewild(last), std::memory_order_acq_rel);
 
```

Comment:
So my investigation of a TSAN error can be found here:
https://gist.github.com/mjp41/3747d988209ca1d0d18a5f09ef9d67c7

I think `acquire` is much safer here.  I think the coherence order on `back` might induce sufficient order to make this actually correct, but moving to `acquire` for now.  We can revisit this choice at a later point if perf is harmed on weaker platforms than x64.